src/Type/MediaItem/MediaItemCreate.php

```diff
@@ -0,0 +1,195 @@
+					},
+				],
+			],
+			'mutateAndGetPayload' => function( $input ) use ( $post_type_object, $mutation_name ) {
+
+				/**
+				 * Throw an exception if there's no input
+				 */
+				if ( ( empty( $post_type_object->name ) ) || ( empty( $input ) || ! is_array( $input ) ) ) {
+					throw new \Exception( __( 'Mutation not processed. There was no input for the mutation or the post_type_object was invalid', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Stop now if a user isn't allowed to create a mediaItem
+				 */
+				if ( ! current_user_can( $post_type_object->cap->create_posts ) ) {
+					throw new \Exception( __( 'Sorry, you are not allowed to create mediaItems', 'wp-graphql' ) );
+				}
+
+				/**
+				 * If the mediaItem being created is being assigned to another user that's not the current user, make sure
+				 * the current user has permission to edit others mediaItems
+				 */
+				if ( ! empty( $input['authorId'] ) && get_current_user_id() !== $input['authorId'] && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+					throw new \Exception( __( 'Sorry, you are not allowed to create mediaItems as this user', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Set the file name, whether it's a local file or from a URL
+				 */
+				$file_name = basename( $input['filePath'] );
+
+				/**
+				 * Check if the download_url method exists and include it if not
+				 * This file also includes the wp_handle_sideload method
+				 */
+				if ( ! function_exists( 'download_url' ) ) {
+					require_once( ABSPATH . 'wp-admin/includes/file.php' );
+				}
+
+				/**
+				 * If the mediaItem file is from a local server, use wp_upload_bits before saving it to the uploads folder
+				 */
+				if ( 'false' === filter_var( $input['filePath'], FILTER_VALIDATE_URL ) ) {
+					$uploaded_file = wp_upload_bits( $file_name, null, file_get_contents( $input['filePath'] ) );
+					$uploaded_file_url = $uploaded_file['url'];
+				} else {
+					$uploaded_file_url = $input['filePath'];
+				}
+
+				/**
+				 * URL data for the mediaItem, timeout value is the default, see:
+				 * https://developer.wordpress.org/reference/functions/download_url/
+				 */
+				$timeout_seconds = 300;
+				$temp_file = download_url( $uploaded_file_url, $timeout_seconds );
+
+				/**
+				 * Build the file data for side loading
+				 */
+				$file_data = [
+					'name'     => $file_name,
+					'type'     => $input['fileType'],
+					'tmp_name' => $temp_file,
+					'error'    => 0,
+					'size'     => filesize( $temp_file ),
+				];
+
+				/**
+				 * Tells WordPress to not look for the POST form fields that would normally be present as
+				 * we downloaded the file from a remote server, so there will be no form fields
+				 * The default is true
+				 */
+				$overrides = [
+					'test_form' => false,
+				];
+
+				/**
+				 * Insert the mediaItem and retrieve it's data
+				 */
+				$file = wp_handle_sideload( $file_data, $overrides );
```

Comment:
Since this function can return either a valid array of file attributes OR an array with an "error" we should account for the error and throw an exception when that happens.